launcher/src/main/java/org/wildfly/core/launcher/DomainCommandBuilder.java

```diff
@@ -626,8 +655,18 @@ public DomainCommandBuilder addProcessControllerJavaOptions(final String... args
      */
     public DomainCommandBuilder addProcessControllerJavaOptions(final Iterable<String> args) {
         if (args != null) {
-            for (String arg : args) {
-                addProcessControllerJavaOption(arg);
+            final Iterator<String> i = args.iterator();
+            for (String arg = i.next(); i.hasNext(); arg = i.next()) {
+                if (Arguments.isModuleSystemArgument(arg)) {
+                    // handle java module system options
+                    if (i.hasNext()) {
+                        this.processControllerJavaOpts.add(Arguments.JavaModuleSystemArgument.of(arg, i.next()));
+                    } else {
+                        // ignore corrupted module system argument
```

Comment:
Same as above, we should just add the arguments and allow the command to fail when executed.